app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
+        :param qs: (optional) queryset with all items of the collection excluding the one
+        being updated.
+        '''
+        logger.debug(
+            'Updating item %s in collection (adjusting start_datetime) %s',
+            item_id,
+            self.collection_name
+        )
+        if old_start_datetime != new_start_datetime:
+            # start_datetime was updated
+            if old_start_datetime == self.cache_start_datetime:
+                # item's old start_datetime was defining left bound of the temporal
+                # extent interval of collection before update
+                if new_start_datetime < old_start_datetime:
+                    # item's start_datetime was shifted to the left (earlier)
+                    self.cache_start_datetime = new_start_datetime
+                else:
+                    # item's start_datetime was shifted to the right (later)
+                    # but was defining the left bound of the temporal extent
+                    # of the collection before
+                    # --> hence the new start_datetime of the collection
+                    # needs to be determinded:
+                    # get earliest start_datetime or none, in case none exists.
+                    # This should be the case, if the item that is currently
+                    # being updated is the only item in the collection or
+                    # no item has a range defined (no start_ and end_datetimes)
+                    # but all do only have properties.datetimes defined.
+                    if bool(qs.filter(properties_start_datetime__isnull=False)):
+                        earliest_start_datetime = qs.filter(
+                            properties_start_datetime__isnull=False
+                        ).earliest('properties_start_datetime').properties_start_datetime
+                    else:
+                        earliest_start_datetime = None
+                    # get earliest datetime, or none in case none exists.
+                    # This should be the case, if the item that is currently
+                    # being updated is the only item in the collection or
+                    # no item has properties.datetime defined but all do have
+                    # only ranged (start_ and end_datetimes defined)
+                    if bool(qs.filter(properties_datetime__isnull=False)):
+                        earliest_datetime = qs.filter(
+                            properties_datetime__isnull=False
+                        ).earliest('properties_datetime').properties_datetime
+                    else:
+                        earliest_datetime = None
```

Comment:
This could also be a function: `earliest_datetime, earliest_start_datetime = get_earliest_datetime(queryset)`

Reply:
Yes, but unfortunately this snippet cannot exactly be reused elsewhere, so it does not make sense to me, to create an extra function for it. Also, the function, to which it currently belongs, is quite short already in current state. So I'll leave it as it is, if OK with you?